### PR TITLE
delete ocsp error response define

### PR DESCRIPTION
**Pull request by simon-xia**
ocsp error response has been defined in  [golang.org/x/crypto/ocsp](https://godoc.org/golang.org/x/crypto/ocsp) yet

```
 var (
         MalformedRequestErrorResponse = []byte{0x30, 0x03, 0x0A, 0x01, 0x01}
         InternalErrorErrorResponse    = []byte{0x30, 0x03, 0x0A, 0x01, 0x02}
         TryLaterErrorResponse         = []byte{0x30, 0x03, 0x0A, 0x01, 0x03}
         SigRequredErrorResponse       = []byte{0x30, 0x03, 0x0A, 0x01, 0x05}
         UnauthorizedErrorResponse     = []byte{0x30, 0x03, 0x0A, 0x01, 0x06}
     )
```

### CODE REVIEW

File: revoke/revoke.go [hunk 1]
```diff
@@ -287,11 +285,16 @@ func sendOCSPRequest(server string, req []byte, issuer *x509.Certificate) (ocspR
...
 	resp.Body.Close()
 
-	if bytes.Equal(body, ocspUnauthorised) {
+	switch {
+	case bytes.Equal(body, ocsp.UnauthorizedErrorResponse):
 		return
-	}
-
-	if bytes.Equal(body, ocspMalformed) {
+	case bytes.Equal(body, ocsp.MalformedRequestErrorResponse):
```
Comment (lziest) [inline revoke/revoke.go]: you can merge multiple case condition together, like case A, B, C, D:, and you can break the line on comma.
If there is a recent update in ocsp package, you may also need to update the vendoring.

**Comment by lziest** [comment]
conflicts, need rebase (and squashed commits)

**Comment by lziest** [comment]
can you rebase again?

File: revoke/revoke.go [hunk 2]
```diff
@@ -288,12 +285,13 @@ func sendOCSPRequest(server string, req []byte, issuer *x509.Certificate) (*ocsp
...
-
-	if bytes.Equal(body, ocspMalformed) {
-		return nil, errors.New("OSCP malformed")
+	switch {
+	case bytes.Equal(body, ocsp.UnauthorizedErrorResponse),
+		bytes.Equal(body, ocsp.MalformedRequestErrorResponse),
+		bytes.Equal(body, ocsp.InternalErrorErrorResponse),
+		bytes.Equal(body, ocsp.TryLaterErrorResponse),
+		bytes.Equal(body, ocsp.SigRequredErrorResponse):
+		return
```
Comment (lziest) [inline revoke/revoke.go]: build failure, since you need to return a list of results.

File: revoke/revoke.go [hunk 3]
```diff
@@ -288,12 +285,13 @@ func sendOCSPRequest(server string, req []byte, issuer *x509.Certificate) (*ocsp
...
-
-	if bytes.Equal(body, ocspMalformed) {
-		return nil, errors.New("OSCP malformed")
+	switch {
+	case bytes.Equal(body, ocsp.UnauthorizedErrorResponse),
+		bytes.Equal(body, ocsp.MalformedRequestErrorResponse),
+		bytes.Equal(body, ocsp.InternalErrorErrorResponse),
+		bytes.Equal(body, ocsp.TryLaterErrorResponse),
+		bytes.Equal(body, ocsp.SigRequredErrorResponse):
+		return nil, err
```
Comment (lziest) [inline revoke/revoke.go]: This is incorrect. err is nil. Sorry we can't merge this if this change doesn't come with tests.

Comment (simon-xia) [inline revoke/revoke.go]: sorry for my mistake, I have fix it

**Comment by lziest** [comment]
LGTM

**Pull request merged**